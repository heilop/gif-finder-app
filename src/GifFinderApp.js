import React, { useState } from 'react';

const GifFinderApp = () => {
  // const categories = ['Mr. Robot', 'The walking Dead', 'Ted Lasso'];
  const [categories, setCategories] = useState(['Mr. Robot', 'The walking Dead', 'Ted Lasso']);

  const handleAdd = (e) => {
    // setCategories([...categories, 'House of Cards']);
    setCategories( categs => [...categs, 'House of Cards']);
  }

  return (
    <>
      <h2>GifFinderApp</h2>
      <br />

      <button onClick={handleAdd}>Add Category</button>

      <ol>
        {
          categories.map(category => {
            return <li key = { category } > { category } </li>
          })
        }
      </ol>
    </>
  );
};

export default GifFinderApp;
