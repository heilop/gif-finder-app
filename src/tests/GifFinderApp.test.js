import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifFinderApp from '../GifFinderApp';


describe('Test in <GifFinderApp />', () => {
  let wrapper = shallow(<GifFinderApp />);

  test('should show <GifFinderApp /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

})
