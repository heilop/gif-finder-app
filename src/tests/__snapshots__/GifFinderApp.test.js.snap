// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifFinderApp /> should show <GifFinderApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifFinderApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <ol />
</Fragment>
`;
